app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,30 @@ module.exports = function(userDao) {
     });
   };
 
+  const signin = function(userData) {
```

Comment:
`signin` doesn't really explain much about what's supposed to happen. This can probably be renamed to something like `authenticateUser` that takes `email` and `password` as arguments. Then the service method can be responsible for:

1. Checking if a user record exists for the given `email` (or `username`)
2. If a user record exists, check to see if the password matches
3. If the password matches then return the `user`